lib/open-api-mocker.js

```diff
@@ -5,53 +5,82 @@ const jsonRefs = require('json-refs');
 const { Parser: OpenApiParser } = require('./openapi');
 const { Parser: ServersParser } = require('./servers');
 const { Parser: PathsParser } = require('./paths');
+const LocalSchemaLoader = require('./schema-loaders/local-loader');
+const OpenAPISchemaInvalid = require('./errors/openapi-schema-invalid-error');
 
 class OpenApiMocker {
 
 	static get defaultPort() {
 		return 5000;
 	}
 
-	constructor({ port, server = 'express' }) {
+	constructor({ port, schema, watch, server = 'express' }) {
 		// eslint-disable-next-line global-require, import/no-dynamic-require
 		const Server = require(`./mocker/${server}/server`);
 
 		this.server = new Server();
 		this.port = port || this.constructor.defaultPort;
+
+		if(schema)
+			this.setSchema(schema);
+
+		this.watch = !!watch;
+		this.alreadyWatching = false;
 	}
 
 	setSchema(schema) {
-		this.schema = schema;
+		if(typeof schema === 'string') {
+			this.schemaLoader = new LocalSchemaLoader(schema);
```

Comment:
Pulling out the schema loading functionality into a new class this does satisfy #27 since it allows both CLI and API usages to share this functionality 👍 .

I fail to see however how a custom schema loader is best implemented with these changes in place.
For instance, if we imagine a RemoteSchemaLoader that fetches a hosted schema via a network request and then periodically polls for changes (i.e. watch functionality).
The implementation of this could pretty much follow the exact interface that the LocalSchemaLoader method uses but there is no way to inject this class. A consumer would have to write this class, call it outside of OpenApiMocker and use the same `setSchema`, `validate` and `mock` method sequence that the CLI currently does outside of this PR.
Whilst this is not too much of an issue, a consumer would have to know this correct sequence to call if they needed to reload the schema based on certain conditions in their own handling of the schema.

I guess another option is to have a constructor parameter similar to the `server` option that allows a custom implementation to be injected or a fallback to the default implementation if not specified. Something like this:
```js
constructor({ schemaLoader }) {
  this.schemaLoader = schemaLoader || new LocalSchemaLoader();
}
```

Reply:
Yes, I didn't add the DI yet. I wanted to avoid generating conflicts with the custom Server implementation, so I was waiting for that to be ready. But definitely the constructor DI will be the way.

Regarding custom schema loaders, for example `RemoteSchemaLoader`, it must only implement a `load()` method that has to take care of schema loading (fetch) and watch feature (polling) in case it supports it.

I'll also write some docs to explain all this _extensions_ that can be made to the core features and how.